engine/evar_kinds.ml

```diff
@@ -21,12 +21,21 @@ type matching_var_kind = FirstOrderPatVar of Id.t | SecondOrderPatVar of Id.t
 
 type subevar_kind = Domain | Codomain | Body
 
+(* maybe this should be a Projection.t *)
+type record_field = { fieldname : Constant.t; recordname : Names.inductive }
+
+type question_mark = {
+     qm_obligation: obligation_definition_status;
+     qm_name: Name.t;
+     qm_record_field: record_field option;
+}
+
```

Comment:
I think you should define somewhere a default record instance
```
let default_question_mark = {
  Evar_kinds.qm_obligation=Evar_kinds.Define true;
  Evar_kinds.qm_name=Anonymous;
  Evar_kinds.qm_record_field=None;
}
```
and replaces all the explicit record creations in the other files by `{ default_question mark with ... }`. The one thing I don't know is where to put this default instance though. Probably not in `Evar_kinds` because this is a type-only file. Probably in `pretyping/glob_ops.ml`.

Reply:
Death to type only files! (I'm serious.) Duplicate said file in a `.ml` and a `.mli` and put the default instance there.

Reply:
I personally disagree, this leads to a shitload of redundancy when describing AST types. Also, as I mentioned already, OCaml relies on the same technique, so it is not as if we were something frowned upon by the language designers...

Reply:
Does OCaml have `.ml` only files? I don't think so. Anyways, I'm not sure that the OCaml compiler code is the right place to look at if you want advice on how to write OCaml code. As if you recommended reading the code of Coq's standard library to know how to write proper Coq code...

Reply:
Fair enough. I prefer having duplicated code than no mli, so @bollu please copy this file into a mli file and add the default record in there.

Reply:
@ppedrot is this notion of a "type-only file" something enforced by the coq codebase style guide? Why would there be such a concept as a "type-only file"? (OCaml noob, trying to understand the motivation for this)

Reply:
@bollu No it's something that has been changing a lot recently. And I personally want to get rid of them, and now I've @ppedrot on board.